ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/TransactionCommitterSpec.scala

```diff
@@ -595,58 +595,125 @@ class TransactionCommitterSpec extends AnyWordSpec with Matchers with MockitoSug
   }
 
   "validateContractKeys" should {
-    "return Inconsistent when a contract key resolves to a different contract ID than submitted by a participant" in {
-      def createNode(contractId: String): Create =
-        create(contractId, signatories = Seq("Alice"), keyAndMaintainer = Some(aKey -> "Alice"))
+    def createNode(contractId: String): Create =
+      create(contractId, signatories = Seq("Alice"), keyAndMaintainer = Some(aKey -> "Alice"))
 
-      def newLookupByKeySubmittedTransaction(found: Boolean): SubmittedTransaction =
-        TransactionBuilder.justSubmitted(
-          txBuilder.lookupByKey(createNode(contractId = s"#$freshContractId"), found)
-        )
+    def freshContractId: String =
+      s"testContractId-${UUID.randomUUID().toString.take(10)}"
+
+    def newLookupByKeySubmittedTransaction(
+        found: Boolean,
+        inRollback: Boolean,
+    ): SubmittedTransaction = {
+      val lookup = txBuilder.lookupByKey(createNode(contractId = s"#$freshContractId"), found)
+      val builder = TransactionBuilder()
+      if (inRollback) {
+        val rollback = builder.add(txBuilder.rollback())
+        builder.add(lookup, rollback)
+      } else {
+        builder.add(lookup)
+      }
+      builder.buildSubmitted()
+    }
+
+    val mykey = {
```

Comment:
If I understand correctly:

```suggestion
    val globalKeyConflictingWithLocalKey = {
```

Reply:
Whether it conflicts with a local key really depends on the test. This is used in a bunch of of different ways across the tests. I’ve just called it `conflictingKey` now which seems vague enough to still be useful.